src/SelectMulti.vue

```diff
@@ -422,35 +431,35 @@
 				aria-multiselectable="true"
 				@mousedown="onMenuMouseDown"
 			>
-			<template v-if="filteredOptions">
-				<div
-					v-for="(option, index) in filteredOptions"
-					:id="`${htmlId}-${index}`"
-					:key="`${option[uniqueIdField as keyof SelectOption]}-${index}`"
-					:class="{
-						'option-current': activeIndex === index,
-						'option-selected': selectedOptions.indexOf(option) > -1,
-						'combo-option': true
-					}"
-					:aria-selected="selectedOptions.indexOf(option) > -1 ? true : false"
-					:aria-label="selectedOptions.indexOf(option) > -1 ? `${option.label} selected` : ''"
-					role="option"
-					@click="onOptionClick(index)"
-					@mousedown="onOptionMouseDown"
-				>
-					<!-- @slot Display individual options via custom template code -->
-					<slot name="option" :option="option">
-						{{  option[labelField as keyof SelectOption] }}
-					</slot>
-				</div>
-			</template>
-			<slot v-if="displayNoResultsMessage" name="no-results">
-				<div class="option-no-results">
-					<span>{{ noResultsMessage }}</span>
-				</div>
-			</slot>
+				<template v-if="filteredOptions">
+					<div
+						v-for="(option, index) in filteredOptions"
+						:id="`${htmlId}-${index}`"
+						:key="`${option[uniqueIdField as keyof SelectOption]}-${index}`"
+						:class="{
+							'option-current': activeIndex === index,
+							'option-selected': selectedOptions.indexOf(option) > -1,
+							'combo-option': true
+						}"
+						:aria-selected="selectedOptions.indexOf(option) > -1 ? true : false"
+						:aria-label="selectedOptions.indexOf(option) > -1 ? `${option.label} selected` : ''"
+						role="option"
+						@click="onOptionClick(index)"
+						@mousedown="onOptionMouseDown"
+					>
+						<!-- @slot Display individual options via custom template code -->
+						<slot name="option" :option="option">
+							{{  option[labelField as keyof SelectOption] }}
+						</slot>
+					</div>
+				</template>
+				<slot v-if="displayNoResultsMessage" name="no-results">
+					<div class="option-no-results">
+						<span>{{ noResultsMessage }}</span>
+					</div>
+				</slot>
 			</div>
-			<div @click="updateMenuState(true)" class="combo-input-icon-block">
+			<div @click="!open ? onIconClick() : null" class="combo-input-icon-block">
```

Comment:
Unfortunately I was unable to use the same solution here that I did on line 418 within the `input`
(`@click="updateMenuState(!open, !open)"`), because when clicking on the 'plus' icon, `updateMenuState` and `onInputBlur` were both triggered, resulting in an extra menu state toggle that cancelled out the desired behavior.

I tried lots of different combinations to get this functionality working and this is the only one that worked for me. Open to feedback if there's a simpler solution that I'm overlooking!

Reply:
I didn't get the idea of `onInputBlur`. Why do we need it?

Reply:
Agree with @ivan-shtyrlyaev-nitka, I don't understand the purpose of `onInputBlur`. Without it the logic would be easier, because now the logic complexity is inreasing significantly, which is not great

Reply:
@ivan-shtyrlyaev-nitka @atsunechka I agree that it's not great how much the complexity is increasing, and that we should keep functionality generic and not project-specific.
`onInputBlur` was already a function previously - this is the change I made:

```diff
onInputBlur() {
				if (this.ignoreBlur) {
					this.ignoreBlur = false
					return
				}

-				this.updateMenuState(false, false)
+				this.updateMenuState(!open, !open)
+				this.ranOnInputBlur = true
			},
```

As I understand it, it functions to close the menu if the user interacts with something outside of the menu. Here's a demo:


https://github.com/user-attachments/assets/a612c3b5-e840-40a8-b59c-34774aab0cd4

Reply:
Hello @cafloyd 
Thanks for the video. That was very helpful. 
So the problem is that no matter what `onInputBlur` runs faster than `updateMenuState`
I assume that blur event is triggered first and we cannot control it.

but we can delay `onInputBlur` (using setTimeout)
and in additional check if it's still open before running `updateMenuState`
![image](https://github.com/user-attachments/assets/a3b3392f-b468-4fbe-bf10-84bda1bc6a87)


and on icon click we simply have the same solution with `updateMenuState` 
![image](https://github.com/user-attachments/assets/e97f9d31-ad3c-4900-89ec-9430938bb0f7)


I know setTimeout is not perfect solution, but in my opinion it's better than adding additional prop 

https://github.com/user-attachments/assets/9c9e9e1c-d288-4355-b598-be1b32b2a44a

Reply:
Thank you @ivan-shtyrlyaev-nitka !! It's not a perfect solution, but I think it's better than the one I came up with so I updated this PR to use your solution: [`c1fcfd2` (#146)](https://github.com/politico/vue-accessible-selects/pull/146/commits/c1fcfd220cf2a5f13a139d680f246bb05ead5c29)